### PR TITLE
added an option in preferences/groups to change the default the hierarchical context

### LINKED ISSUE #9141: Add default hierachical context in group creation
Thanks for your hard work developing this very sefull software. 

I like all my groups to be of union type as hierarchical context and It is a bit tedious to have to change this type each time I create a new group. 

Then, in my opinion, it would be nice to be able to change the default hierarchical context for new group in the the parameters once for all.

### PR DESCRIPTION
**Pull request by roryoday**
Fixes #9141 

added an option in preferences/groups to change the default the hierarchical context. Options are Union/Intersection/Independent.  

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by ThiloteE** [comment]
To ease organizational workflows I have linked the pull-request to the issue with syntax as described in https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

> <h2 id="linking-a-pull-request-to-an-issue-using-a-keyword">Linking a pull request to an issue using a keyword</h2>
> <p>You can link a pull request to an issue by using a supported keyword 
> in the pull request's description or in a commit message. The pull 
> request <strong>must be</strong> on the default branch.</p>
> <ul>
> <li>close</li>
> <li>closes</li>
> <li>closed</li>
> <li>fix</li>
> <li>fixes</li>
> <li>fixed</li>
> <li>resolve</li>
> <li>resolves</li>
> <li>resolved</li>
> </ul>
> <p>If you use a keyword to reference a pull request comment in another 
> pull request, the pull requests will be linked. Merging the referencing 
> pull request also closes the referenced pull request.</p>
> <p>The syntax for closing keywords depends on whether the issue is in the same repository as the pull request.</p>

**Comment by Siedlerchr** [comment]
Please have a look at  the failing checkstyle tests or reviewdog

File: src/main/java/org/jabref/model/groups/GroupHierarchyType.java [hunk 1]
```diff
@@ -29,4 +29,15 @@ public static GroupHierarchyType getByNumberOrDefault(int type) {
...
+    public String getFormattedName() {
+        if (this == INDEPENDENT) {
+            return "Independent";
+        } else if (this == REFINING) {
+            return "Intersection";
+        }
+        else {
+            return "Union";
+        }
+    }
```
Comment (calixtus) [inline src/main/java/org/jabref/model/groups/GroupHierarchyType.java:42]: This should be implemented as an constructor argument for the enum e.g.  `INDEPENDENT("Independent")` and be localized. Open/closed design principle.

File: src/test/java/org/jabref/gui/groups/GroupCreateTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    void groupContextCorrect() {
+        String groupName = "group";
+        ExplicitGroup group = new ExplicitGroup(groupName, GroupHierarchyType.INDEPENDENT, ',');
+        BibEntry entry = new BibEntry();
+        databaseContext.getDatabase().insertEntry(entry);
+        GroupNodeViewModel model = new GroupNodeViewModel(databaseContext, stateManager, taskExecutor, group, new CustomLocalDragboard(), preferencesService);
+        model.addEntriesToGroup(databaseContext.getEntries());
+        // make sure group default hierarchy preference does not override new group
+        assertEquals(model.getGroupNode().getGroup().getHierarchicalContext(),GroupHierarchyType.INDEPENDENT);
+    }
```
Comment (calixtus) [inline src/test/java/org/jabref/gui/groups/GroupCreateTest.java:69]: I really don't get what you want to test here?
The constructor of GroupNodeViewModel?
This test seems superfluous to me, especially it does not seem to be related to any code change in the PR. The thing that should be tested is if someone clicks in the UI an "add new group" the default GroupHierarchyType is selected.

**Review by calixtus — CHANGES_REQUESTED** [review]
Hi, thanks for your interest in JabRef programming. I looked quickly into your changes. Please configure checkstyle according to our code style standards. Please look into our Getting Started guide here (https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace.html),

Plase integrate the requested changes, so we can discuss merging.

**Comment by ThiloteE** [comment]
closing this, as there is a lack of response and to reduce the number of open pull-requests.
@roryoday  Feel free to reopen, if you think you will continue to work on this.

**Pull request closed without merging**